addons/account/models/account_invoice.py

```diff
@@ -385,12 +386,55 @@ def get_view_id(xid, name):
         return super(AccountInvoice, self).fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
 
     @api.multi
+    def _get_edi_attachments(self):
+        ''' Get the EDI documents in the filestore or generates new ones.
+        '''
+        self.ensure_one()
+        country_code = self.company_id.country_id.code
+        func_filenames = 'get_edi_filenames_' + country_code
+        func_attachment = 'generate_edi_' + country_code
+        if not hasattr(self, func_filenames) or self.state not in ('open', 'paid') or self.type not in ('out_invoice', 'out_refund'):
+            return []
+        if self.sent:
+            domain = [
+                ('res_id','=', self.id),
+                ('res_model', '=', self._name),
+                ('name', 'in', getattr(self, func_filenames)())]
+            return self.env['ir.attachment'].search(domain)
+        else:
+            if not hasattr(self, func_attachment):
+                raise UserError(_('Unable to generate EDI: Method %s not found' % func_attachment))
```

Comment:
In EDI mechanism it is not a good approach have Raises because frequently you can have the document in odoo without the EDI itself, it is better have a mail.message (or another log system)